packages/calcite-components/src/components/segmented-control/segmented-control.tsx

```diff
@@ -350,26 +350,40 @@ export class SegmentedControl
     });
   }
 
-  //--------------------------------------------------------------------------
-  //
-  //  Private Methods
-  //
-  //--------------------------------------------------------------------------
+  private handleSelectedItem(): void {
+    const { items } = this;
 
-  onLabelClick(): void {
-    this.setFocus();
+    const lastChecked = items.filter((item) => item.checked).pop();
+
+    if (lastChecked) {
+      this.selectItem(lastChecked);
+    } else if (items[0]) {
+      items[0].tabIndex = 0;
+    }
   }
 
-  private getItems(): HTMLCalciteSegmentedControlItemElement[] {
-    return Array.from(this.el.querySelectorAll("calcite-segmented-control-item"));
+  private handleDefaultSlotChange = (event: Event): void => {
+    const items = slotChangeGetAssignedElements(event).filter(
```

Comment:
According to [the doc](https://developers.arcgis.com/calcite-design-system/components/segmented-control/#slots), we only support `calcite-segmented-control-items`, so we shouldn't need to filter items.

Reply:
I think we should leave this incase users slot tooltips, divs or (html comments?). Its safer to know we only want the segmented-control-items and this is the default slot. Otherwise there could potentially be a regression since we are currently querying items with querySelector.